source/credentials_provider_sts.c

```diff
@@ -239,6 +253,36 @@ static bool s_on_node_encountered_fn(struct aws_xml_parser *parser, struct aws_x
     return true;
 }
 
+/* errors that mean something screwy was going on at the networking layer. */
+static inline bool s_is_transient_error(int error_code) {
+    return error_code == AWS_ERROR_HTTP_CONNECTION_CLOSED || error_code == AWS_ERROR_HTTP_STREAM_CLOSED ||
```

Comment:
`AWS_ERROR_HTTP_STREAM_CLOSED ` this error code is no longer existed. Not sure why the checks passed.

Reply:
Looks like this error code is specific to H2 and has been moved to h2 error code. Shall we remove it from here? Or shall we change it to AWS_ERROR_HTTP_SERVER_CLOSED, which can also be a networking layer error.